### PR TITLE
Adds unit tests/stubs to AccountServiceRemote

### LINKED ISSUE #6991: Improve the AccountServiceRemote tests
Let's add some test coverage and stub out the network calls for the following endpoints:
- `me`
- `me/sites`
- `is-available/email`
- `is-available/username`
- `auth/send-login-email`

### PR DESCRIPTION
**Pull request by bummytime**
Fixes #6991 This PR is the beginning a series of PRs that will add missing unit tests to the `*Remote` classes. For this class (`AccountServiceRemoteREST`), all of the possible endpoint responses and status codes are encapsulated using mock data and [OHHTTPStubs](https://github.com/AliSoftware/OHHTTPStubs). 

To test: Get the latest from this branch and make sure all of the unit tests are passing.

Needs review: @astralbodies

### CODE REVIEW

File: WordPress/WordPressTest/AccountServiceRemoteRESTTests.swift [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+    let setSiteVisibilityFailureMockFilename        = "me-sites-visibility-failure.json"
+    let setSiteVisibilityBadJsonFailureMockFilename = "me-sites-visibility-bad-json-failure.json"
+    let isEmailAvailableSuccessMockFilename         = "is-available-email-success.json"
+    let isEmailAvailableFailureMockFilename         = "is-available-email-failure.json"
+    let isUsernameAvailableSuccessMockFilename      = "is-available-username-success.json"
+    let isUsernameAvailableFailureMockFilename      = "is-available-username-failure.json"
+    let requestLinkSuccessMockFilename              = "auth-send-login-email-success.json"
+    let requestLinkNoSuchUserFailureMockFilename    = "auth-send-login-email-no-user-failure.json"
+    let requestLinkInvalidClientFailureMockFilename = "auth-send-login-email-invalid-client-failure.json"
+    let requestLinkInvalidSecretFailureMockFilename = "auth-send-login-email-invalid-secret-failure.json"
```
Comment (astralbodies) [inline WordPress/WordPressTest/AccountServiceRemoteRESTTests.swift]: @jleandroperez you see the damage you've done here?

File: WordPress/WordPressTest/AccountServiceRemoteRESTTests.m [hunk 2]
```diff
@@ -1,54 +0,0 @@
-#import <OCMock/OCMock.h>
```
Comment (astralbodies) [inline WordPress/WordPressTest/AccountServiceRemoteRESTTests.m]: Bye bye Objc!

File: WordPress/WordPressTest/RemoteTestCase.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import Foundation
+import XCTest
+import OHHTTPStubs
+@testable import WordPress
+
+class RemoteTestCase: XCTestCase {
```
Comment (astralbodies) [inline WordPress/WordPressTest/RemoteTestCase.swift]: I like having a base test case class for this - lots of good stuff.

File: WordPress/WordPressTest/RemoteTestCase.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        ContextManager.overrideSharedInstance(nil)
+        contextManager.mainContext.reset()
+        contextManager = nil
+        restApi = nil
+        OHHTTPStubs.removeAllStubs()
+    }
+
+    // MARK: - Helpers
+
+    func stubRemoteResponse(_ endpoint: String, filename: String, contentType: String, status: Int32 = 200) {
```
Comment (astralbodies) [inline WordPress/WordPressTest/RemoteTestCase.swift]: It doesn't HAVE to but consider putting this into an extension for `RemoteTestCase`.

File: WordPress/WordPressTest/Test Data/me-sites-visibility-bad-json-failure.json [hunk 5]
```diff
@@ -0,0 +1,2 @@
+{
+    "success": true
```
Comment (astralbodies) [inline WordPress/WordPressTest/Test Data/me-sites-visibility-bad-json-failure.json]: lol I was just going to say this was malformed JSON

File: WordPress/WordPressTest/PeopleRemoteTests.swift [hunk 6]
```diff
@@ -14,49 +21,29 @@ class PeopleServiceTests: XCTestCase {
     let deleteFollowerSuccessMockFilename   = "people-delete-follower-success.json"
     let deleteViewerFailureMockFilename     = "people-delete-viewer-failure.json"
     let deleteViewerSuccessMockFilename     = "people-delete-viewer-success.json"
-    let contentTypeJson                     = "application/json"
-    let timeout                             = TimeInterval(1000)
 
     // MARK: - Properties
 
-    var restApi: WordPressComRestApi!
     var remote: PeopleRemote!
```
Comment (astralbodies) [inline WordPress/WordPressTest/PeopleRemoteTests.swift]: I know it's not part of this PR but could we rename the subject class (and related classes) to `PeopleServiceRemote` to be consistent?

File: WordPress/WordPressTest/PeopleRemoteTests.swift [hunk 7]
```diff
@@ -14,49 +21,29 @@ class PeopleServiceTests: XCTestCase {
...
-            return fixture(filePath: stubPath!, status: status, headers: ["Content-Type" as NSObject: self.contentTypeJson as AnyObject])
-        }
-    }
-
-
     // MARK: - Tests
 
     func testValidateInvitationWithInvalidUsernameFails() {
-        let expect = expectation(description: "Send Invite")
+        let expect = expectation(description: "Validate invite failure")
```
Comment (astralbodies) [inline WordPress/WordPressTest/PeopleRemoteTests.swift]: Thanks for taking the time to give unique descriptions on the expectations!

File: WordPress/WordPressTest/RemoteTestCase.swift [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+    let contentTypeJson     = "application/json"
+    let contentTypeJS       = "text/javascript;charset=utf-8"
+    let contentTypeHTML     = "application/html"
+    let timeout             = TimeInterval(1000)
+
+    // MARK: - Properties
+
+    var restApi: WordPressComRestApi!
+    var contextManager: TestContextManager!
```
Comment (astralbodies) [inline WordPress/WordPressTest/RemoteTestCase.swift]: I'm wondering if calling this `var testContextManager` would be better so superclasses don't look like they're trying to use the default one based upon first glance of the name.

Comment (bummytime) [inline WordPress/WordPressTest/RemoteTestCase.swift]: I will change!

File: WordPress/WordPressTest/PeopleRemoteTests.swift (hunk 6 above)
Comment (bummytime) [inline WordPress/WordPressTest/PeopleRemoteTests.swift]: Yeps!

File: WordPress/WordPressTest/AccountServiceRemoteRESTTests.swift (hunk 1 above)
Comment (jleandroperez) [inline WordPress/WordPressTest/AccountServiceRemoteRESTTests.swift]: YAY!!!

**Comment by bummytime** [comment]
Ok @astralbodies. All of your PR comments have been addressed. Also, I added the logic to fail if any other endpoint is called besides the one under test (the issue we talked about offline has been resolved). Take another look when you have a moment! 

Thanks! 😄

**Pull request merged**